### PR TITLE
VC-4229: added methods for MsgSetDefaultWallet

### PR DESCRIPTION
**Pull request by at-kh**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/wallets/set_default_wallet.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/forbole/bdjuno/v2/database/types"
+)
+
+// SaveDefaultWallets - inserts messages into the "vipcoin_chain_wallets_set_default_wallet" table
+func (r Repository) SaveDefaultWallets(messages ...*walletstypes.MsgSetDefaultWallet) error {
+	if len(messages) == 0 {
+		return nil
+	}
+
+	tx, err := r.db.BeginTxx(context.Background(), &sql.TxOptions{})
```
Comment (alex-bodnar) [inline database/vipcoin/chain/wallets/set_default_wallet.go]: Можно упростить до чего то такого:

	query := `INSERT INTO vipcoin_chain_wallets_set_default_wallet 
			(creator, address) 
			VALUES 
			(:creator, :address)`

	if _, err := r.db.NamedExec(query, toSetDefaultWalletsDatabase(msg...)); err != nil {
		return err
	}

Получиться без транзакций

File: modules/vipcoin/chain/wallets/handle_msg.go [hunk 2]
```diff
@@ -70,3 +72,21 @@ func (m *Module) handleMsgSetStates(msg *typeswallets.MsgSetWalletState) error {
...
+
+	wallets, err := m.walletsRepo.GetWallets(filter.NewFilter().SetArgument(FieldAddress, msg.Address))
+	if err != nil {
+		return err
+	}
+
+	if len(wallets) != 1 {
+		return typeswallets.ErrInvalidAddressField
+	}
+
```
Comment (alex-bodnar) [inline modules/vipcoin/chain/wallets/handle_msg.go]: Может какое-то поле в wallets обновить нужно?

**Comment by alex-bodnar** [comment]
Удали пожалуйста Copyright в файлах которые ты добавлял, он не нужен так как не мы разработчики bdjuno

File: modules/vipcoin/chain/wallets/handle_msg.go (hunk 2 above)
Comment (at-kh) [reply modules/vipcoin/chain/wallets/handle_msg.go]: Wallet.Creator не существует, а обновлять нечего

File: database/vipcoin/chain/wallets/set_default_wallet.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		}
+	}
+
+	return nil
+}
+
+// GetDefaultWallets - get the given messages from the "vipcoin_chain_wallets_set_default_wallet" table
+func (r Repository) GetDefaultWallets(filter filter.Filter) ([]*walletstypes.MsgSetDefaultWallet, error) {
+	query, args := filter.Build("vipcoin_chain_wallets_set_default_wallet",
+		`creator, address`)
```
Comment (vasylNaumenko) [inline database/vipcoin/chain/wallets/set_default_wallet.go]: query, args := filter.Build(
		tableName,
		m.FieldID, m.FieldPollsItemID, m.FieldPlayerID, m.FieldAccountAddress,
		m.FieldPlayerVotes, m.FieldStatus, m.FieldCreatedAt, m.FieldUpdatedAt,
	)

по аналогии

File: database/schema/12-wallets.sql [hunk 4]
```diff
@@ -10,40 +10,40 @@ CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_wallets (
...
+    creator     TEXT    NOT NULL,                           -- set wallet kind creator
+    address     TEXT    NOT NULL,                           -- set wallet kind address
+    kind        INT                                         -- set wallet kind kind
 );
 
 CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_set_wallet_state (
-    creator     TEXT    NOT NULL,                                   -- set wallet state creator
-    address     TEXT    NOT NULL,                                   -- set wallet state address
-    state       INT                                                 -- set wallet state state
+    creator     TEXT    NOT NULL,                           -- set wallet state creator
```
Comment (vasylNaumenko) [inline database/schema/12-wallets.sql]: -- message creator

File: database/schema/12-wallets.sql [hunk 5]
```diff
@@ -10,40 +10,40 @@ CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_wallets (
...
+    address     TEXT    NOT NULL,                           -- set wallet state address
```
Comment (vasylNaumenko) [inline database/schema/12-wallets.sql]: -- target wallet address

File: database/schema/12-wallets.sql [hunk 6]
```diff
@@ -10,40 +10,40 @@ CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_wallets (
...
+    state       INT                                         -- set wallet state state
```
Comment (vasylNaumenko) [inline database/schema/12-wallets.sql]: -- new state for target wallet

File: database/schema/12-wallets.sql [hunk 7]
```diff
@@ -10,40 +10,40 @@ CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_wallets (
...
 CREATE TABLE IF NOT EXISTS vipcoin_chain_wallets_create_wallet_with_balance (
-    creator             TEXT        NOT NULL,                               -- create wallet with balance creator
-    address             TEXT        NOT NULL,                               -- create wallet with balance address
-    account_address     TEXT        NOT NULL,                               -- create wallet with balance account address
-    kind                INT,                                                -- create wallet with balance kind
-    state               INT,                                                -- create wallet with balance state
-    extras              JSONB,                                              -- create wallet with balance extras
-    default_status      BOOLEAN,                                            -- create wallet with balance default
-    balance             JSONB                                               -- create wallet with balance balance
+    creator             TEXT        NOT NULL,               -- create wallet with balance creator
```
Comment (vasylNaumenko) [inline database/schema/12-wallets.sql]: убрать тафтологию в комментах, попытаться описать смысл переменной, если это поле creator - то создатель чего?
поле адрес - адрес чего? и так далее.

File: modules/vipcoin/chain/wallets/handle_msg.go (hunk 2 above)
Comment (alex-bodnar) [reply modules/vipcoin/chain/wallets/handle_msg.go]: // handleMsgSetKinds allows to properly handle a handleMsgSetKinds
func (m *Module) handleMsgSetDefaultWallet(msg *typeswallets.MsgSetDefaultWallet) error {
	if err := m.walletsRepo.SaveDefaultWallets(msg); err != nil {
		return err
	}

	address, err := sdk.AccAddressFromBech32(msg.Address)
	if err != nil {
		return typeswallets.ErrInvalidAddressField
	}

	targetWallet, err := m.walletsRepo.GetWallets(filter.NewFilter().SetArgument(FieldAddress, address))
	if err != nil {
		return err
	}

	if targetWallet[0].Address == "" {
		return typeswallets.ErrNotFoundWallet
	}

	account, err := m.accountsRepo.GetAccounts(filter.NewFilter().SetArgument(FieldAddress, targetWallet[0].AccountAddress))
	if err != nil {
		return err
	}

	for _, walletAddr := range account[0].Wallets {
		w, err := m.walletsRepo.GetWallets(filter.NewFilter().SetArgument(FieldAddress, walletAddr))
		if err != nil {
			return err
		}
		if !w[0].Default {
			continue
		}
		w[0].Default = false
		if err := m.walletsRepo.SaveWallets(w[0]); err != nil {
			return err
		}
		break
	}

	targetWallet[0].Default = true
	return m.walletsRepo.SaveWallets(targetWallet[0])
}

File: database/vipcoin/chain/wallets/set_default_wallet.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetDefaultWallets - get the given messages from the "vipcoin_chain_wallets_set_default_wallet" table
+func (r Repository) GetDefaultWallets(filter filter.Filter) ([]*walletstypes.MsgSetDefaultWallet, error) {
+	query, args := filter.Build("vipcoin_chain_wallets_set_default_wallet", types.FieldCreator, types.FieldAddress)
```
Comment (vasylNaumenko) [inline database/vipcoin/chain/wallets/set_default_wallet.go:32]: вынести в переменную репозитория tableName

**Comment by vasylNaumenko** [comment]
lgtm

**Comment by alex-bodnar** [comment]
LGTM

**Pull request merged**